src/openloco/gui.cpp

```diff
@@ -0,0 +1,198 @@
+#include "gui.h"
+#include "graphics/colours.h"
+#include "interop/interop.hpp"
+#include "objects/interface_skin_object.h"
+#include "objects/objectmgr.h"
+#include "openloco.h"
+#include "tutorial.h"
+#include "windowmgr.h"
+
+namespace windowmgr = openloco::ui::windowmgr;
+using window_type = openloco::ui::window_type;
+using namespace openloco::interop;
+
+namespace openloco::gui
+{
+
+    loco_global_array<openloco::ui::widget, 64, 0x00509c20> _mainWindowWidgets;
+
+    loco_global<uint8_t, 0x00508F14> _screen_flags;
+
+    // 0x00438A6C
+    void init()
+    {
+        const int uiWidth = 288;
+        const int uiHeight = 512;
+
+        _mainWindowWidgets[0].bottom = uiHeight;
+        _mainWindowWidgets[0].right = uiWidth;
+        auto window = openloco::ui::windowmgr::create_window(
+            window_type::main,
+            0,
+            0,
+            uiWidth,
+            uiHeight,
+            0x1,
+            (void*)0x004FA1F4);
+        window->widgets = _mainWindowWidgets;
+
+        // viewport_create
+        registers regs;
+        regs.edx = 0x17FF17FF;
+        regs.eax = (window->y << 16) | window->x;
+        regs.ebx = (window->height << 16) | window->width;
+        regs.ecx = 0x1e00000;
+        regs.esi = (uint32_t)window;
+        addr<0x00e3f0b8, int32_t>() = 0; // gCurrentRotation
+        call(0x004ca2d0, regs);
```

Comment:
Should just create a wrapper for this then you can have a function with zero use of registers.